shared/src/types/ethereum_events.rs

```diff
@@ -5,24 +5,43 @@ use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
 
 use crate::proto::MultiSigned;
 use crate::types::address::Address;
+use crate::types::hash::Hash;
 use crate::types::token::Amount;
 
+/// Each event has a nonce set by the Ethereum smart contract that emitted it.
+#[derive(
+    Debug, PartialEq, Eq, Clone, BorshSerialize, BorshDeserialize, BorshSchema,
+)]
+pub struct EthereumEventNonce(u64);
+
 /// An Ethereum event to be processed by the Anoma ledger
-#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, BorshSchema)]
+#[derive(
+    Debug, PartialEq, Eq, Clone, BorshSerialize, BorshDeserialize, BorshSchema,
+)]
 pub enum EthereumEvent {
     /// Event transferring batches of ether from Ethereum to wrapped ETH on
     /// Anoma
-    TransfersToNamada(Vec<TransferToNamada>),
+    TransfersToNamada(Vec<TransferToNamada>, EthereumEventNonce),
```

Comment:
I'm not sure how best to include the nonce

Reply:
I think we should refactor the type something like
```
pub enum RawEvent {
    TransfersToNamada(Vec<TransferToNamada>),
    // all other variants which also have a nonce
    ...
}

pub struct EthereumEvent {
    pub nonce: Nonce,
    pub event: RawEvent,
}
```
Although I don't particularly like the name `RawEvent`.

Reply:
:+1: I've updated it to be like in your comment. I'm not sure about the name `RawEvent` either but we can always change it later